### PR TITLE
Processing data in model

### PR DESCRIPTION
**Pull request by Fikerte-T**
In this project, I implemented the following
- Created model classes `User`, `Post`, `Comment`, and `Like`
- Setup associations between the models
- Used the models to insert data and used other CRUD methods to find, update and delete entities
- Create custom methods to models

          In User
                   - A method that returns the 3 most recent posts for a given user.
          In Post
                   - A method that updates the posts counter for a user.
                   - A method that returns the 5 most recent comments for a given post.
          In Comment
                   - A method that updates the comments counter for a post.
          In Like
                   - A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :comments
+  has_many :likes
```
Comment (dasileker) [inline app/models/user.rb]: - [ ] Please add the foreign keys 'author_id' for posts and comments and likes.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :user
```
Comment (dasileker) [inline app/models/post.rb]: - [ ] Also the belong to the user should have been `author` and class_name to the user

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
```
Comment (dasileker) [inline app/models/like.rb]: - [ ] same here the as the post.rb

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
```
Comment (dasileker) [inline app/models/comment.rb]: - [ ] same here as the post.rb.

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+  def update_posts_counter
+    user.increment(:posts_counter)
+  end
+
+  def recent_comments
+    comments.limit(5).order(created_at: :desc)
+  end
+end
```
Comment (dasileker) [inline app/models/post.rb]: - 

- [ ] also We should be private for the increment method, and have an after_save

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @Fikerte-T,** 

### **Good job so far!**
There are some issues that you still need to work on to go to the next project but you are almost there!

## **Highlights**

- Linters are passing ✅.
- Good readme file ✅.
- Nice PR ✅.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @Fikerte-T 👋🏻, 

Good job making the requested changes 💪🏻!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

thats-how-its-done-allison-janney


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**